### ISSUE TITLE
Issue a useful error when calling a function with too many arguments

### ISSUE THREAD

**Issue by RyanCavanaugh** [Microsoft/TypeScript#6045]
``` ts
'x'.substr(1, 2, 3);
```

Gives you this:

> error TS2346: Supplied parameters do not match any signature of call target

This is unhelpful. We should elaborate with an error about arity.

**Comment by mhegazy** [comment]
:+1:

**Comment by bowdenk7** [comment]
Thoughts from a user on this matter:

![image](https://cloud.githubusercontent.com/assets/820883/13021727/96dff92c-d191-11e5-9f19-ac257a1fb5df.png)

**Comment by mhegazy** [comment]
A fix here should also handle generic type argument count mismatch, see #101.


```ts
function f<T, U>(x: T, y: U): T { return null; }
var r1 = f<number>(1, '');
``` 
Expected:

> Signature expected 2 type arguments, got 1 instead.

Actual:

> Supplied parameters do not match any signature of call target.

**Comment by aleksey-bykov** [comment]
is it a dup of `Supplied parameters do not match any signature of call target.`?

**Comment by DanielRosenwasser** [comment]
Fixed by #15861

![image](https://user-images.githubusercontent.com/972891/28948611-acfcd2ac-786b-11e7-9f33-5b17d3dbdd4a.png)
